design/datamover.md

```diff
@@ -0,0 +1,208 @@
+
+2. Volume Snapshots: Storage providers provide with volume snapshot capabilities via plugins. These snapshots are crash consistent but there is no guarantee that
+   these snapshots are durable. These snapshots are generally stored locally in the cluster storage, so if the cluster breaks then the snapshots are of no use. Velero also supports
+   CSI volume snapshots, but as there is no data movement provided by Velero for these CSI snapshots they are not durable and dependent on the storage provider.
+
+
+Related Issue: https://github.com/vmware-tanzu/velero/issues/3229
+
+## Goals
+- To be able to backup CSI snapshots to object storage
+- To be able to recreate CSI snapshots under a different storage provider
+- To provide an extensible data mover solution which should allow third parties to plugin their own datamover implementation
+- To provide a Datamover example PoC
+
+## Non Goals
+- Maintain 3rd party data mover implementations
+- Adding a status watch controller to Velero
+- To support incremental backups (future scope, dependent on standard CSI support for snapshot differentials)
+- To allow plug-ins for services like encryption, compression etc. (future scope)
+
+## High-Level Design ([VolSync](https://github.com/backube/volsync) as a working Proof of Concept)
+The goal of this high level design is to explain a working PoC of moving CSI snapshots where-in a datamover (VolSync) works in conjunction with Velero via a DataMover plugin (will be installed when the DataMover flag is set to true).
+
+The following design proposal is for adding a datamover feature to Velero which can be integrated with various user implemented datamovers.
+
+![](datamover.png)
+
+The design components and CRs that we are introducing are as follows:
+
+1. DataMoverType(DMT): This is a cluster scoped Custom Resource that will have details about the data mover provisioner.The provisioner will be responsible
+   for creating a DataMover plugin that will create a `DataMoverBackup` CR & `DataMoverRestore` CR. They will also be responsible for implementing the logic
+   for DataMoverBackup & DataMoverRestore controller that corresponds to their data mover. The `DataMoverType` name will be referenced in `DataMoverBackup`
+   & `DataMoverRestore` CRs. This will help in selecting the data mover implementation during runtime. If the `DataMoverType` name is not defined, then the
+   default `DataMoverType` will be used, which in this case will be `VolSync`
+
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  name: datamovertype.velero.io
+spec:
+  provisioner: <VolSync>
+```
+
+2. DataMoverBackup(DMB): Assuming that the flag to enable DataMover is set to true during Velero install, when a velero backup is created, it triggers DataMover plugin
```

Comment:
IMO the movement should be optional.
I don't think it's correct that we always create a DMB to kickoff the data movement for each backup, each pvc or snapshot.
For example on EKS the snapshot is moved under the hood so the DMB is no longer needed.